src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -1247,6 +1255,45 @@ public function hasSelectModifier($modifier)
         return in_array($modifier, $this->selectModifiers);
     }
 
+    /**
+     * @return null|\Propel\Runtime\ActiveQuery\Lock Get read lock value.
+     */
+    public function getLock(): ?Lock
+    {
+        return $this->lock;
+    }
+
+    /**
+     * Apply a read lock to be used to retrieve rows.
+     *
+     * @see self::SHARED
+     * @see self::EXCLUSIVE
+     *
+     * @param string   $lockType
+     * @param string[] $tableNames
+     * @param bool     $noWait
+     *
+     * @return $this Modified Criteria object (for fluent API)
+     */
+    public function withLock(string $lockType, array $tableNames = [], bool $noWait = false): self
+    {
+        $this->lock = new Lock($lockType, $tableNames, $noWait);
+
+        return $this;
+    }
+
+    /**
+     * Retrieve rows without any read locking.
+     *
+     * @return $this Modified Criteria object (for fluent API)
+     */
+    public function withoutLock(): self
```

Comment:
Please avoid chainable typehints. $this must stay without
Not sure if the sniffer already reports this.

Reply:
You mean this?
```suggestion
    public function withoutLock()
```
sniffer didn't report this